v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/transforms/RangeCountDoFn.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.source.reader.io.jdbc.uniformsplitter.transforms;
+
+import static org.apache.beam.sdk.util.Preconditions.checkStateNotNull;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.uniformsplitter.range.Range;
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.uniformsplitter.range.RangePreparedStatementSetter;
+import java.io.Serializable;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLTimeoutException;
+import javax.annotation.Nullable;
+import javax.sql.DataSource;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class RangeCountDoFn extends DoFn<Range, Range> implements Serializable {
+
+  /**
+   * We set a millisecond precision timeout for the DB query. Additionally, we also set a timeout at
+   * the level of prepared statement. Setting the timeout at the level of prepared statement covers
+   * for older versions of MySQL that don't honour the magic comment for timeout (note that 5.7.4+
+   * support the select level timeout). Since the timeout at the level of prepared statement is in
+   * seconds, we add 1500 milliseconds to cover for the rounding off error and giving 500
+   * milliseconds for db to vm network latency.
+   */
+  private static final long TIMEOUT_GRACE_MILLIS = 1500;
+
+  private static final Logger logger = LoggerFactory.getLogger(RangeCountDoFn.class);
+  private final SerializableFunction<Void, DataSource> dataSourceProviderFn;
+  private final long timeoutMillis;
+
+  private final String countQuery;
+
+  private final long numColumns;
+
+  @JsonIgnore private transient @Nullable DataSource dataSource;
+
+  RangeCountDoFn(
```

Comment:
is this meant to be default ?

Reply:
Sorry I am not getting what we are referring to `this` here?

On the current line 58 where the comment is anchored, Its the constructor of the DoFn

Reply:
I meant default access - not public or private

Reply:
Got you. Thanks.
It is package private and not constructed outside the package.